lib/base/io-engine.cpp

```diff
@@ -153,3 +153,14 @@ void Timeout::Cancel()
 	boost::system::error_code ec;
 	m_Timer.cancel(ec);
 }
+
+/**
+ * Cancels the timeout and waits for the own coroutine to finish.
+ *
+ * @param yc Yield Context for ASIO
+ */
```

Comment:
On its own, this doc comment misses context, i.e. there's no other doc comment that mentions a coroutine.

I think it would be more helpful if it instead stated the effects. Like that it waits for the involved resources to be destroyed, which includes waiting for the callback if it was already triggered.